homeassistant/components/screenlogic/water_heater.py

```diff
@@ -0,0 +1,121 @@
+"""Support for a ScreenLogic Water Heater."""
+import logging
+
+from screenlogicpy.const import HEAT_MODE
+
+from homeassistant.components.water_heater import (
+    SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    WaterHeaterEntity,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+
+from . import ScreenlogicEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_FEATURES = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
+
+HEAT_MODE_NAMES = HEAT_MODE._names  # pylint: disable=protected-access
+
+MODE_NAME_TO_MODE_NUM = {
+    HEAT_MODE_NAMES[num]: num for num in range(len(HEAT_MODE_NAMES))
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+    entities = []
+    data = hass.data[DOMAIN][config_entry.entry_id]
+
+    for body in data["devices"]["water_heater"]:
+        entities.append(ScreenLogicWaterHeater(data["coordinator"], body))
+    async_add_entities(entities, True)
+
+
+class ScreenLogicWaterHeater(ScreenlogicEntity, WaterHeaterEntity):
+    """Represents the heating functions for a body of water."""
+
+    @property
+    def name(self) -> str:
+        """Name of the water heater."""
+        ent_name = self.body["heat_status"]["name"]
+        return f"{self.gateway_name} {ent_name}"
+
+    @property
+    def state(self) -> str:
+        """State of the water heater."""
+        return HEAT_MODE.GetFriendlyName(self.body["heat_status"]["value"])
+
+    @property
+    def min_temp(self) -> float:
+        """Minimum allowed temperature."""
+        return self.body["min_set_point"]["value"]
+
+    @property
+    def max_temp(self) -> float:
+        """Maximum allowed temperature."""
+        return self.body["max_set_point"]["value"]
+
+    @property
+    def current_temperature(self) -> float:
+        """Return water temperature."""
+        return self.body["last_temperature"]["value"]
+
+    @property
+    def target_temperature(self) -> float:
+        """Target temperature."""
+        return self.body["heat_set_point"]["value"]
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement."""
+        if self.config_data["is_celcius"]["value"] == 1:
+            return TEMP_CELSIUS
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_operation(self) -> str:
+        """Return operation."""
+        return HEAT_MODE.GetFriendlyName(self.body["heat_mode"]["value"])
+
+    @property
+    def operation_list(self):
+        """All available operations."""
+        return HEAT_MODE_NAMES
+
```

Comment:
<img width="380" alt="Screen Shot 2021-03-15 at 8 14 40 AM" src="https://user-images.githubusercontent.com/663432/111201167-96e9df80-8566-11eb-9d09-b60addcfa6b9.png">

Can we filter out modes that the aren't supported (ex `Solar`).  Also, I think `Don't Change` can be removed

Reply:
Define supported. Supported by the API? Supported by that specific configuration of the pool controller? Supported as a standard mode in Home Assistant?
I'll look into removeing `Don't Change`.

Reply:
> Supported by the user's specific configuration of the pool controller 

The four pools I tested it on don't have Solar so I wasn't expect it to show up.

Reply:
I'm wondering if we can tell from `equipment_flags` if it has solar or not.

Reply:
Maybe https://github.com/parnic/node-screenlogic/blob/main/messages/SLControllerConfigMessage.js#L9

Reply:
https://github.com/parnic/node-screenlogic/blob/main/messages/SLControllerConfigMessage.js#L93

Looks like its `equipment_flags & 0x1`

Reply:
I see. 
I had been taking notes here: https://github.com/dieselrabbit/screenlogicpy/blob/master/notes/equip_flags.txt but since I couldn't account for all options (`heater`) from the equipment flags, I was avoiding it all together. 
Would you consider it better to have all options consistently available and the user be responsible for setting the correct one (because the controller will still take the setting) or support the likely majority of use cases, but in a technically broken way, because `heater` will always be available, even if there is no heater?
To me this also has an impact on the future addition of color lights color/shows, as I feel even farther away from understanding how color-capable being available is identified.

Reply:
I think it is fine for the heater to always appear if we can't figure out which flag it is linked to.  It can always be improved later.  Since we can we figure out which flag controls if solar is present, we should hide that when it is not.

Ideally the lighting moves to be implemented on the `light` platform once more functionality is added (colors/etc).  I can provide config dumps on the 4 pools I have access to if that helps.

Reply:
Ok. I'll update the `water_heater` to remove options when we _know_ the equipment doesn't support it.
Config dumps would be great. Even better would be also knowing what major features they have so I can try to align them to equipment_flags.

As for the future `light`, because there appears to be no full RGB setable lighting from Pentair (only specific colors), it seems currently I'd have to implement all the shows **and** colors as effects, unless we're able to specify which colors go into the light color picker?

Reply:
comit [cf9b5e9](https://github.com/home-assistant/core/pull/47933/commits/cf9b5e9f5dd688f094ba071237135a7ab0745f14) should do this.